api/src/main/java/com/flipkart/poseidon/api/APILegoSet.java

```diff
@@ -110,7 +109,33 @@ public Buildable getBuildable(Request request) throws LegoSetException, ElementN
         return buildable;
     }
 
-    private String[] getKeysForTrie(String url) {
+    public static List<KeyWrapper<String>> getKeysForTrie(String url) {
+        return getKeysForTrie(getKeysArrayForTrie(url));
+    }
+
+    public static void resolveKeys(List<KeyWrapper<String>> wrappers) {
+        for (KeyWrapper<String> keyWrapper : wrappers) {
+            if (keyWrapper.key.startsWith("{") && keyWrapper.key.endsWith("}")) {
+                keyWrapper.key = null;
+                keyWrapper.wildCard = true;
+            } else if (keyWrapper.key.startsWith("*") && keyWrapper.key.endsWith("*")) {
+                keyWrapper.key = null;
+                keyWrapper.wildPath = true;
+            }
+        }
+    }
+
+    public static List<KeyWrapper<String>> getResolvedKeysForTrie(String[] keys) {
+        List<KeyWrapper<String>> wrappers = Arrays.stream(keys).map(KeyWrapper::new).collect(Collectors.toList());
+        resolveKeys(wrappers);
+        return wrappers;
+    }
+
+    public static List<KeyWrapper<String>> getKeysForTrie(String[] keys) {
+        return Arrays.stream(keys).map(KeyWrapper::new).collect(Collectors.toList());
+    }
+
+    private static String[] getKeysArrayForTrie(String url) {
```

Comment:
There are more than required functions here. Can we keep just the following

public static List<KeyWrapper<String>> getWrapperKeysForTrie(String Url) // We can change tests to construct string url rather than array. So this will be used while adding to trie and in tests
private String[] getKeysForTrie(String Url) // This will be used while fetching from trie

getWrapperKeysForTrie(String url) will call getKeysForTrie(url), convert string array to resolved List<KeyWrapper> (just a single for loop instead of stream + for loop).

No other public and static methods would be required.